skiplang/prelude/src/stdlib/collections/persistent/SortedMap.sk

```diff
@@ -325,16 +325,25 @@ base class .SortedMap<+K: Orderable, +V> uses  //  extends KeyedSequence<K, V>
     value: V2,
     f: (V2, V2) -> V2,
   ): SortedMap<K2, V2>
-  | Nil() -> SortedMap::node(key, value, Nil(), Nil())
-  | Node _ ->
+  | Nil() -> SortedMap::singleton(key, value)
+  | n @ Node _ ->
     l: SortedMap<K2, V2> = this.left;
     r: SortedMap<K2, V2> = this.right;
     k: K2 = this.key;
     v: V2 = this.value;
     compare(key, this.key) match {
-    | LT() -> SortedMap::balance(k, v, l.setWith(key, value, f), r)
-    | EQ() -> SortedMap::node(k, f(v, value), l, r)
-    | GT() -> SortedMap::balance(k, v, l, r.setWith(key, value, f))
+    | LT() ->
+      newL = l.setWith(key, value, f);
+      if (phys_eq(newL, l)) this else SortedMap::balance(k, v, newL, r)
+    | EQ() ->
+      newV = f(v, value);
+      if (phys_eq(newV, v)) this else {
```

Comment:
Is this the problematic call, or are there others? This one could perhaps be replaced with the `==` from the `Equality` trait on `V` without being slow.

Reply:
All the calls to `phys_eq` are problematic for `value class` objects (as explained in #900).

I tried to check `contains` before adding elements but it didn't change the memory footprint, so there must be another issue.

Reply:
Ok, I guess the thrust of my question is: is this the only call to phys_eq added in this PR for a value class?

Reply:
Yes it is.

By the way, there is currently no constraint that `V` implements `Equality`